exec/java-exec/src/test/java/org/apache/drill/exec/physical/impl/join/TestMergeJoinAdvanced.java

```diff
@@ -253,4 +259,75 @@ public void testDrill4196() throws Exception {
       .baselineValues(6000*800L)
       .go();
   }
+
+  private void buildFile(String fileName, String[] data, File testDir) throws IOException {
+    try(PrintWriter out = new PrintWriter(new FileWriter(new File(testDir, fileName)))) {
+      for (String line : data) {
+        out.println(line);
+      }
+    }
+  }
+
+  @Test
+  public void testMergeLeftJoinWithEmptyTable() throws Exception {
+    ClusterFixtureBuilder builder = ClusterFixture.builder(dirTestWatcher);
+    File testDir = null;
+    try {
+      ClusterFixture cluster = builder.build();
+      ClientFixture client = cluster.clientFixture();
+      testDir = dirTestWatcher.getTmpDir();
+      cluster.defineWorkspace("dfs", "data", testDir.getAbsolutePath(), "json");
+      buildFile("dept.json", new String[0], testDir);
+      QueryBuilder query = client.queryBuilder().sql("select * from cp.`employee.json` emp left outer join dfs.data.`dept.json` as dept on dept.manager = emp.`last_name`");
+      assert(query.futureSummary().get().recordCount() == 1155);
+    } catch (RuntimeException ex) {
+      throw ex;
+    } finally {
+      if (testDir != null) {
+        testDir.delete();
+      }
+    }
+  }
+
+  @Test
+  public void testMergeInnerJoinWithEmptyTable() throws Exception {
+    ClusterFixtureBuilder builder = ClusterFixture.builder(dirTestWatcher);
+    File testDir = null;
+    try {
+      ClusterFixture cluster = builder.build();
+      ClientFixture client = cluster.clientFixture();
+      testDir = dirTestWatcher.getTmpDir();
+      cluster.defineWorkspace("dfs", "data", testDir.getAbsolutePath(), "json");
+      buildFile("dept.json", new String[0], testDir);
+      QueryBuilder query = client.queryBuilder().sql("select * from cp.`employee.json` emp inner join dfs.data.`dept.json` as dept on dept.manager = emp.`last_name`");
+      assert(query.futureSummary().get().recordCount() == 0);
```

Comment:
How could this test ensure that we did, indeed, stop the readers from reading extra records? (That is, that the `kill()` calls worked?)